rasa/core/lock_store.py

```diff
@@ -310,15 +312,20 @@ def _create_from_endpoint_config(
         # this is the default type if no lock store type is set
 
         lock_store: LockStore = InMemoryLockStore()
-    elif endpoint_config.type == "redis":
+    elif endpoint_config.type == "redis" and not endpoint_config.kwargs.get(
+        CONCURRENT_KEY, False
+    ):
         rasa.shared.utils.io.raise_deprecation_warning(
-            "'RedisLockStore' is deprecated and will be removed in Rasa Open "
-            "Source 4.0.0. Please migrate to the enhanced 'ConcurrentRedisLockStore'.",
+            "'RedisLockStore' now supports concurrent message handling. "
+            "You must set the 'concurrent' key to True to activate this improvement. ",
```

Comment:
will `concurrent: false` be deprecated in 4.0.0? If so I think the deprecation warning should stay, unless we plan to keep both even in 4.0?

Reply:
The only reason I suggested to keep both was because of the UX research synthesis that pointed to what is perceived as abrupt deprecations in rasa-oss. That plus the cons to managing the migration online [here](https://www.notion.so/rasa/LockStore-Improvements-a3e948bb4aae4b6ba35521fe2cc69aa0#04ea8c6f9aa44a279389ba4ea50daf76) made me think maybe we should discuss if we should deprecate at all.

Reply:
let's resolve this and keep both.